components/brave_wallet/browser/hd_key.cc

```diff
@@ -136,6 +142,192 @@ std::unique_ptr<HDKey> HDKey::GenerateFromPrivateKey(
+      return nullptr;
+    }
+    derived_key = SymmetricKey::DeriveKeyFromPasswordUsingScrypt(
+        SymmetricKey::AES, password,
+        std::string(salt_bytes.begin(), salt_bytes.end()), (size_t)*n,
+        (size_t)*r, (size_t)*p, 512 * 1024 * 1024, (size_t)*dklen * 8);
+    if (!derived_key) {
+      VLOG(1) << __func__ << ": scrypt derivation failed";
+      return nullptr;
+    }
+  } else {
+    VLOG(0) << __func__
+            << ": kdf is not supported. (Only support pbkdf2 and scrypt)";
+    return nullptr;
+  }
+
+  const auto* mac = crypto->FindStringKey("mac");
+  if (!mac) {
+    VLOG(0) << __func__ << ": missing mac";
+    return nullptr;
+  }
+  const auto* ciphertext = crypto->FindStringKey("ciphertext");
+  if (!ciphertext) {
+    VLOG(0) << __func__ << ": missing ciphertext";
+    return nullptr;
+  }
+  std::vector<uint8_t> ciphertext_bytes;
+  if (!base::HexStringToBytes(*ciphertext, &ciphertext_bytes)) {
+    VLOG(1) << __func__ << ": invalid ciphertext";
+    return nullptr;
+  }
+
+  std::vector<uint8_t> mac_verification_input(
+      derived_key->key().end() - *dklen / 2, derived_key->key().end());
+  mac_verification_input.insert(mac_verification_input.end(),
+                                ciphertext_bytes.begin(),
+                                ciphertext_bytes.end());
+  // verify password
+  std::vector<uint8_t> mac_verification(KeccakHash(mac_verification_input));
+  if (base::ToLowerASCII(base::HexEncode(mac_verification)) != *mac) {
+    VLOG(0) << "password does not match";
+    return nullptr;
+  }
+
+  const auto* cipher = crypto->FindStringKey("cipher");
+  if (!cipher) {
+    VLOG(0) << __func__ << ": missing cipher";
+    return nullptr;
+  }
+  if (*cipher != "aes-128-ctr") {
+    VLOG(0) << __func__
+            << ": AES-128-CTR is the minimal requirement of version 3";
+    return nullptr;
+  }
+
+  std::vector<uint8_t> iv_bytes;
+  const auto* iv = crypto->FindStringPath("cipherparams.iv");
+  if (!iv) {
+    VLOG(0) << __func__ << ": missing cipherparams.iv";
+    return nullptr;
+  }
+  if (!base::HexStringToBytes(*iv, &iv_bytes)) {
+    VLOG(1) << __func__ << ": invalid iv";
+    return nullptr;
+  }
+
+  std::vector<uint8_t> private_key;
+  std::unique_ptr<SymmetricKey> decryption_key = SymmetricKey::Import(
+      SymmetricKey::AES, derived_key->key().substr(0, *dklen / 2));
+  if (!decryption_key) {
+    VLOG(1) << __func__ << ": raw key has to be 16 or 32 bytes for AES import";
+    return nullptr;
+  }
+  Encryptor encryptor;
+  if (!encryptor.Init(decryption_key.get(), Encryptor::Mode::CTR,
+                      std::vector<uint8_t>())) {
+    VLOG(0) << __func__ << ": encryptor init failed";
+    return nullptr;
+  }
+  if (!encryptor.SetCounter(iv_bytes)) {
+    VLOG(0) << __func__ << ": encryptor set counter failed";
+    return nullptr;
+  }
+
+  if (!encryptor.Decrypt(ciphertext_bytes, &private_key)) {
+    VLOG(0) << __func__ << ": encryptor decrypt failed";
```

Comment:
are these logs for debugging? doesn't seem necessary to keep them in release builds.

Reply:
It is not for debugging, if users import the json file and then get error, we would ask them to turn on the vlog to figure out what went wrong